dipy/viz/ui.py

```diff
@@ -0,0 +1,562 @@
+
+        self.handle_events()
+
+        self.on_left_mouse_button_pressed = lambda i_ren, obj, element: None
+        self.on_left_mouse_button_drag = lambda i_ren, obj, element: None
+        self.on_right_mouse_button_pressed = lambda i_ren, obj, element: None
+        self.on_right_mouse_button_drag = lambda i_ren, obj, element: None
+        self.on_mouse_hover = lambda i_ren, obj, element: None
+
+    def get_actors(self):
+        """ Returns the actors that compose this UI component. """
+        msg = "Subclasses of UI must implement `get_actors(self)`."
+        raise NotImplementedError(msg)
+
+    def add_to_renderer(self, ren):
+        """ Allows UI objects to add their own props to the renderer.
+        Parameters
+        ----------
+        ren : renderer
+        """
+        ren.add(*self.get_actors())
+
+        # Get a hold on the current interactor style.
+        iren = ren.GetRenderWindow().GetInteractor().GetInteractorStyle()
+
+        for callback in self._callbacks:
+            if not isinstance(iren, CustomInteractorStyle):
+                msg = ("The ShowManager requires `CustomInteractorStyle` in"
+                       " order to use callbacks.")
+                raise TypeError(msg)
+
+            iren.add_callback(*callback, args=[self])
+
+    def add_callback(self, prop, event_type, callback, priority=0):
+        """ Adds a callback to a specific event for this UI component.
+        Parameters
+        ----------
+        prop : vtkProp
+        event_type : event code
+        callback : function
+        priority : int
+        """
+        # Actually since we need an interactor style we will add the callback
+        # only when this UI component is added to the renderer.
+        self._callbacks.append((prop, event_type, callback, priority))
+
+    def set_center(self, position):
+        """ Sets the center of the UI component
+        Parameters
+        ----------
+        position : (float, float)
+        """
+        msg = "Subclasses of UI must implement `set_center(self, position)`."
+        raise NotImplementedError(msg)
+
+    def set_visibility(self, visibility):
+        """ Sets visibility of this UI component. """
+        for actor in self.get_actors():
+            actor.SetVisibility(visibility)
+
+    def handle_events(self):
+        self.add_callback("LeftButtonPressEvent", self.left_button_click_callback)
+        self.add_callback("LeftButtonReleaseEvent", self.left_button_release_callback)
+        self.add_callback("RightButtonPressEvent", self.right_button_click_callback)
+        self.add_callback("RightButtonReleaseEvent", self.right_button_release_callback)
+        self.add_callback("MouseMoveEvent", self.mouse_move_callback)
+
+    @staticmethod
+    def left_button_click_callback(i_ren, obj, self):
+        self.left_button_state = "clicked"
+        i_ren.event.abort()
+
+    @staticmethod
+    def left_button_release_callback(i_ren, obj, self):
```

Comment:
Oh! I see, it is because of how we actually called the callback functions (https://github.com/nipy/dipy/pull/1140/files#diff-a1072548746ff0b78f876c2efab5bd0eR286).

We should use `self` as an argument name since it has a special meaning. Let's call it `ui_object`.